text/0000-destructure-drop-structs.md

```diff
@@ -0,0 +1,125 @@
+- Feature Name: Destructure structs implementing Drop
+- Start Date: 2017-07-08
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Allow destructuring of structs that implement Drop.
```

Comment:
It would be nice if the RFC actually said what is destructuring before proceeding to motivatio.
Especially given that the problem is not about destructuring in patterns, but about moving out individual fields from in general. It can be done through any field access.
```
struct NonCopy;
let ds = DropStruct { individual_field: NonCopy, other_individual_field: NonCopy };
let f = ds.individual_field; // Moving out without destructuring.
// f is initialized, DROPPED
// ds is uninitialized, NOT DROPPED
// ds.individual_field is uninitialized, NOT DROPPED
// ds.other_individual_field is still initialized, DROPPED
```

Reply:
@petrochenkov 
I did not intend to allow moving out individual fields for precisely this reason.
(Will clarify this in the rfc)